### PR TITLE
Python 3 port

### PR DESCRIPTION
**Pull request by jamadden**

### CODE REVIEW

File: src/zope/app/dependable/dependency.py [hunk 1]
```diff
@@ -16,6 +16,8 @@
 on an object having dependencies. It raises an exception if it's the
 case.
 """
+from __future__ import print_function, absolute_import, division
+
```
Comment (mgedmin) [inline src/zope/app/dependable/dependency.py]: I don't see any print statements or divisions in this module, and I'd be very much surprised if any of the `from zope.x.y.z` imports could be affected by `absolute_import`.

File: setup.py [hunk 2]
```diff
@@ -17,55 +17,64 @@
...
+          'Programming Language :: Python :: 3.5',
+          'Programming Language :: Python :: 3.6',
           'Programming Language :: Python :: Implementation :: CPython',
+          'Programming Language :: Python :: Implementation :: PyPy',
           'Natural Language :: English',
           'Operating System :: OS Independent',
           'Topic :: Internet :: WWW/HTTP',
-          'Framework :: Zope3'],
-      url='http://pypi.python.org/pypi/zope.app.dependable',
+          'Framework :: Zope3'
```
Comment (mgedmin) [inline setup.py]: I like trailing commas when the closing bracket is on the next line.

File: src/zope/app/dependable/interfaces.py [hunk 3]
```diff
@@ -12,9 +12,8 @@
 #
 ##############################################################################
 """Dependable framework interfaces
-
-$Id$
 """
+from __future__ import print_function, absolute_import, division
```
Comment (mgedmin) [inline src/zope/app/dependable/interfaces.py]: These don't seem required either.

**Review by mgedmin — APPROVED** [review]
LGTM.

(The pointless `__future__` directives don't actually hurt anything.)

**Review by mgedmin — APPROVED** [review]

**Comment by jamadden** [comment]
I didn't actually hit 100% coverage. I'm working on that real quick.

**Comment by jamadden** [comment]
OK, we actually hit 100% coverage this time. There's one branch I couldn't find a scenario that would trigger so I suspect it's dead code. But that may depend on the versions of dependencies in use...

File: src/zope/app/dependable/__init__.py [hunk 4]
```diff
@@ -84,7 +85,11 @@ def _make_relative(self, path):
             path = canonicalPath(path)
             if path.startswith(self.pp):
                 path = path[self.pplen:]
-                while path.startswith("/"):
+                while path.startswith("/"): # pragma: no cover
+                    # We should not be able to get here. canonicalPath
+                    # doesn't allow trailing / in a path segment, and we
+                    # already cut off the whole length of the parent, which
+                    # we guaranteed to begin and end with a /
                     path = path[1:]
```
Comment (mgedmin) [inline src/zope/app/dependable/__init__.py]: Also, this is a really manual way to spell `path = path.lstrip("/")`.

Comment (jamadden) [reply src/zope/app/dependable/__init__.py]: Ahh, good point, I didn't think of that. That seems like a safe change.

File: src/zope/app/dependable/__init__.py [hunk 5]
```diff
@@ -85,12 +85,15 @@ def _make_relative(self, path):
...
-                    # already cut off the whole length of the parent, which
-                    # we guaranteed to begin and end with a /
-                    path = path[1:]
+                # Now, the path should not actually begin with a /.
+                # canonicalPath doesn't allow trailing / in a path
+                # segment, and we already cut off the whole length of
+                # the parent, which we guaranteed to begin and end
+                # with a /. But it's possible that older dependencies
+                # than we test with could produce this scenario, so we
+                # leave it for BWC.
```
Comment (mgedmin) [inline src/zope/app/dependable/__init__.py:94]: BWC = backwards compatibility?  First time I see this acronym.

(We used to use BBB in Zope-land, which doesn't stand for anything, and I guess was just a version of XXX, but for backwards compatibility support code.)

**Review by mgedmin — APPROVED** [review]
LGTM

File: src/zope/app/dependable/__init__.py (hunk 5 above)
Comment (jamadden) [reply src/zope/app/dependable/__init__.py:94]: > BWC = backwards compatibility? First time I see this acronym.

Indeed. I guess that's what I "grew up" using? Maybe it's more common in Java, or maybe it was just the company culture. In the code I have handy, I find hits for it in :
- zope.proxy
- zope.session
- zc.intid
- relstorage
- gevent 

But to be fair, some of those *may* be me.

> We used to use BBB in Zope-land, which doesn't stand for anything

That's really good to know. I've wracked my brain trying, and failing, to figure out what that meant.

**Comment by jamadden** [comment]
Thanks. A PyPI release of this will be needed to move forward on https://github.com/zopefoundation/zope.app.container/pull/1

**Comment by mgedmin** [comment]
You now have PyPI rights to zope.app.dependable.

**Pull request merged**